src/modules/iotjs_module_buffer.c

```diff
@@ -687,6 +687,73 @@ JS_FUNCTION(ByteLength) {
 }
 
 
+JS_FUNCTION(FromArrayBuffer) {
+  if (jargc < 1 || !jerry_value_is_arraybuffer(jargv[0])) {
+    return jerry_create_undefined();
+  }
+  jerry_length_t offset = 0;
+  jerry_length_t length = jerry_get_arraybuffer_byte_length(jargv[0]);
+
+  if (jargc >= 2) {
+    jerry_value_t offset_num = jerry_value_to_number(jargv[1]);
+
+    if (jerry_value_is_error(offset_num)) {
+      return offset_num;
+    }
+
+    double offset_value = jerry_get_number_value(offset_num);
+    if (isnan(offset_value)) {
+      offset_value = 0;
+    }
+    jerry_release_value(offset_num);
+
+    if (offset_value == length) {
+      offset = length;
+    } else if (offset_value >= 0 && offset_value < length) {
+      offset = (jerry_length_t)offset_value;
+    } else {
+      return JS_CREATE_ERROR(RANGE, "'offset' is out of bounds");
+    }
+  }
+
+  length -= offset;
+
+  if (jargc >= 3) {
+    if (jerry_value_is_error(jargv[2])) {
+      return length;
+    }
+
+    double length_value = (double)length;
```

Comment:
The `length_value` is only used in the `if` block below, Could we move this there?